pkg/compact/retention.go

```diff
@@ -24,6 +25,17 @@ func ApplyDefaultRetentionPolicy(ctx context.Context, logger log.Logger, bkt obj
 			return errors.Wrap(err, "download metadata")
 		}
 
+		retentionDuration, ok := retentionByResolution[ResolutionLevel(m.Thanos.Downsample.Resolution)]
+		if !ok {
+			level.Info(logger).Log("msg", "retention for resolution undefined", "resolution", m.Thanos.Downsample.Resolution)
```

Comment:
I don't think that is realy needed. We already notified about this and this one will just spam a lot. Let's log only in case of retention being applied.